src/vanilla/store.ts

```diff
@@ -519,14 +519,83 @@ const buildStore = (
     }
   }
 
+  const peekAtomState = <Value>(
+    pending: Pending | undefined,
+    atom: Atom<Value>,
+    dirtyAtoms?: Set<AnyAtom>,
+  ): AtomState<Value> => {
+    const atomState = getAtomState(atom)
```

Comment:
note: we cannot skip recomputing the atom value each time because we do not update dependencies. This may make deep peek a deal-breaker.